include/openthread-types.h

```diff
@@ -336,6 +336,23 @@ typedef enum
 } otDeviceRole;
 
 /**
+ * This structure represents the MAC layer counter.
+ */
+typedef struct otMacCounter
+{
+    uint32_t mTxData;
+    uint32_t mTxBeacon;
+    uint32_t mTxBeaconRequest;
+    uint32_t mTxDataRetry;
+    uint32_t mTxDataErrAck;
+    uint32_t mTxErrCca;
+    uint32_t mRxData;
+    uint32_t mRxBeacon;
+    uint32_t mRxBeaconRequest;
+    uint32_t mRxErrSec;
+} otMacCounter;
```

Comment:
Thanks for working on this!  These diagnostic counters will be very useful.  Please add doxygen so that it's easy to know what the counters mean.

Minimally, we should be able to extract the following information:
- all packet transmissions
- cca failures
- acknowledgements received in response to a data transmission
- transmitted data, beacons, beacon requests, and data poll messages
- all packet receptions
- all received packets with FCS errors
- all received packets with security errors
- all other received packets that were filtered
- received data, beacons, beacon requests, and data poll messages